iris-mpc-cpu/src/execution/hawk_main/scheduler.rs

```diff
@@ -50,7 +73,41 @@ pub fn schedule(
                 }
             })
         })
-        .collect_vec()
+        .collect_vec();
+
+    Schedule {
+        n_eyes,
+        n_requests,
+        n_rotations,
+        batches,
+    }
+}
+
+impl Schedule {
+    pub fn organize_results<T>(
+        &self,
+        mut results: HashMap<TaskId, T>,
+    ) -> Result<BothEyes<VecRequests<VecRots<T>>>> {
+        assert_eq!(self.n_eyes, 2);
+
+        Ok([0, 1].map(|i_eye| {
+            (0..self.n_requests)
+                .map(|i_request| {
+                    (0..self.n_rotations)
+                        .map(|i_rotation| {
+                            let task = Task {
+                                i_eye,
+                                i_request,
+                                i_rotation,
+                            };
+                            results.remove(&task.id()).expect("missing result")
```

Comment:
Would it make sense to follow the auditors' suggestion to avoid panics here?

Reply:
Done.